ext/drreg/drreg.c

```diff
@@ -209,6 +232,9 @@ spill_reg(void *drcontext, per_thread_t *pt, reg_id_t reg, uint slot, instrlist_
         dr_spill_slot_t DR_slot = (dr_spill_slot_t)(slot - ops.num_spill_slots);
         dr_save_reg(drcontext, ilist, where, reg, DR_slot);
     }
+    if (TEST(DRREG_HANDLE_MULTI_PHASE_SLOT_RESERVATIONS, pt->bb_props)) {
+        instr_set_note(instr_get_prev(where), (void *)(ptr_uint_t)slot);
```

Comment:
Notes need to have conventions to avoid conflicts.  Use drmgr_reserve_note_range() to get unique values, and only put notes on instructions this code is inserting: I guess that last part is the case here.  It might be worth a comment that this assumes instr_get_prev grabs one of the (potentially multiple for dr_save_reg?) spill instrs to avoid someone placing code in between.

Reply:
> It might be worth a comment that this assumes instr_get_prev grabs one of the (potentially multiple for dr_save_reg?) spill instrs to avoid someone placing code in between.

Done

> Notes need to have conventions to avoid conflicts. Use drmgr_reserve_note_range() to get unique values,

Done; I added it like drmgr does it. But I don't think I understand how note conflicts are avoided by `drmgr_reserve_note_range()`. I only understand that we prevent conflicts between clients on note _value ranges_, but it's still possible for clients (and core DR too) to overwrite note field set by each other right?

Reply:
What is the order between callbacks of multiple clients? Based on https://dynamorio.org/page_deploy.html#multi_client, I understand that each client completes all callbacks before letting other clients run. So, does this mean there cannot be any conflicts in the note field itself?

Therefore, we only need to worry about misinterpretation of note values residual from previous clients (which is solved by note value reservation)?

Reply:
Generally it is multiple components of one client, rather than multiple clients.  I.e., multiple libraries like these extension libraries, plus the end client itself.  drmgr adds priorities used to order handlers: see e.g. DRMGR_PRIORITY_INSERT_DRREG_LOW.

Usually note fields are limited to label instructions, and the creator of the label sets the note.  Thus there are no write conflicts because a new note is put in place by creating a new label.  Reserving note ranges lets later consumers of the data iterate the list and read note values of labels and know which means what.  It might be better to use labels here.

Reply:
I used a label instr. I found that labels can have data areas, and using that seemed much cleaner way to put this sort of info. So now I use a special note to mark label instrs that contain spill slot usage info in their data area.